reader/reader/core/reader_service.py

```diff
@@ -12,26 +25,127 @@
 
 @service_as_factory
 class ReaderService:
-    def get(self, request: GetRequest):
-        return request.fqid
 
-    def get_many(self, request: GetManyRequest):
-        pass
+    connection: ConnectionHandler
+    database: ReadDatabase
+
+    def get(self, request: GetRequest) -> Model:
+        with self.database.get_context():
+            deleted = self.database.is_deleted(request.fqid, request.position)
+            if (
+                deleted
+                and request.get_deleted_models == DeletedModelsBehaviour.NO_DELETED
+            ):
+                raise ModelDoesNotExist(request.fqid)
+            if (
+                not deleted
+                and request.get_deleted_models == DeletedModelsBehaviour.ONLY_DELETED
+            ):
+                raise ModelNotDeleted(request.fqid)
+
+            if request.position:
+                model = self.database.build_model_ignore_deleted(
+                    request.fqid, request.position
+                )
+                model = self.apply_mapped_fields(model, request.mapped_fields)
+            else:
+                model = self.database.get(request.fqid, request.mapped_fields)
```

Comment:
I was wondering, why this works: you are trusting the default argument of get_many in the backend service. I suggest staying consistent with get_many here: Move the detection to the backend's get method and add the DeletedModelsBehaviour as an argument there. Also do a custom sql query: You can safe one query if you get the model and deletet state at the same time and then decide, if you want to throw an error.

Reply:
Without position, that's fine. With position, if you'd want to do it with one query, you'd have to fetch all events and check if the last one is a deleted event, which is probably more overhead than to just execute 2 queries.